store/mockstore/unistore/cophandler/closure_exec.go

```diff
@@ -62,6 +62,39 @@ func mapPkStatusToHandleStatus(pkStatus int) tablecodec.HandleStatus {
 	return tablecodec.HandleDefault
 }
 
+func getExecutorList(dagReq *tipb.DAGRequest) ([]*tipb.Executor, error) {
+	if len(dagReq.Executors) > 0 {
+		return dagReq.Executors, nil
+	}
+	// convert TiFlash executors tree to executor list
+	executors := make([]*tipb.Executor, 0, 3)
+	currentExec := dagReq.RootExecutor
+	for currentExec.Tp != tipb.ExecType_TypeTableScan {
+		executors = append(executors, currentExec)
+		switch currentExec.Tp {
+		case tipb.ExecType_TypeTopN:
+			currentExec = currentExec.TopN.Child
+		case tipb.ExecType_TypeStreamAgg, tipb.ExecType_TypeAggregation:
+			currentExec = currentExec.Aggregation.Child
+		case tipb.ExecType_TypeLimit:
+			currentExec = currentExec.Limit.Child
```

Comment:
no join or selection?

Reply:
Join is not supported yet, selection is handled in https://github.com/pingcap/tidb/blob/master/store/mockstore/unistore/cophandler/closure_exec.go#L83